binary/core.py

```diff
@@ -132,7 +131,9 @@ def convert_units(n, unit=BYTE, to=None, si=False, exact=False):
         raise ValueError(f'{unit} is not a valid binary unit.')
 
     # Always work with bytes to simplify logic.
-    n *= Decimal(unit) if exact else unit
+    if exact:
+        n = Decimal(n)
```

Comment:
@ofek 

If the concern is the floating point precision way off to the right, we can coerce this to a str first.

 ```suggestion
        n = Decimal(str(n))
```

If that is the concern, I can add a test for it.